Fix double include of sha256.h causing a build error

danielinux opened the pull request:
Per ZD 12243: When compiling a HW driver with sha256 support (e.g. DCP) with `HAVE_HASHDRBG` and without `NO_RNG` or `CUSTOM_RAND_GENERATE_BLOCK`, a build error occurs:

```
In file included from ./wolfssl/wolfcrypt/port/nxp/dcp_port.h:26:0,
                 from ./wolfssl/wolfcrypt/sha256.h:135,
                 from ./wolfssl/wolfcrypt/random.h:106,
                 from wolfcrypt/src/random.c:53:
./wolfssl/wolfcrypt/tfm.h:835:65: error: unknown type name ‘WC_RNG’; did you mean ‘DT_REG’?
 MP_API int  mp_prime_is_prime_ex(mp_int* a, int t, int* result, WC_RNG* rng);
                                                                 ^~~~~~
                                                                 DT_REG
```

Fixed by removing explicit include of sha256.h in HASHDBRG case.

Edit: except when required by `--enable-smallstackcache` as Jenkins pointed out.

dgarske reviewed (changes requested):
This fix feel more like a workaround. This issue will still exist if you build `WOLFSSL_IMXRT_DCP` with `WOLFSSL_SMALL_STACK_CACHE`. 

Is the real issue here that `dcp_port.h` is defining its own `wc_Sha256`? Most crypto HW ports put their context into the actual `sha256.h` `wc_Sha256`. For example adding `dcp_handle_t` and `dcp_hash_ctx_t`.

If you would like to merge this patch and plan for a proper cleanup of this at a later time that is fine, just let me know.

danielinux commented:
Thanks for the review, David. This mechanism was borrowed from older ports (see [ti-hash](https://github.com/wolfSSL/wolfssl/blob/master/wolfssl/wolfcrypt/port/ti/ti-hash.h#L55)). I agree that the suggested approach is cleaner, I am going to produce a better fix.

danielinux commented:
I've changed the driver as suggested.

`wc_Sha256` and `wc_Sha` are no longer re-defined in `dcp_port.h`. Instead, the actual struct associated is modified accordingly.

I have reverted the workaround in random.h. The build error with HDRBG went away.

I have tested all the functions on the target and the dcp driver still works fine.

dgarske commented on wolfcrypt/src/port/nxp/dcp_port.c:
```diff
@@ -35,6 +35,8 @@
 #include <wolfssl/wolfcrypt/sha.h>
 #include <wolfssl/wolfcrypt/sha256.h>
 #include <wolfssl/wolfcrypt/error-crypt.h>
+#include "fsl_device_registers.h"
```
Please move these into the `#ifdef WOLFSSL_IMXRT_DCP` section, otherwise will produce build errors on systems including all .c files.

danielinux replied:
fixed, thanks

josepho0918 commented on wolfcrypt/src/port/nxp/dcp_port.c:
```diff
@@ -49,6 +51,15 @@
 #include "fsl_debug_console.h"
 #include "fsl_dcp.h"
 
+#ifdef USE_FAST_MATH
+    #include <wolfssl/wolfcrypt/tfm.h>
+#elif defined WOLFSSL_SP_MATH
+    #include <wolfssl/wolfcrypt/sp_int.h>
+#else
+    #include <wolfssl/wolfcrypt/integer.h>
+#endif
```
```suggestion
#include <wolfssl/wolfcrypt/integer.h>
```

danielinux replied:
Thanks, I've removed the includes entirely now.
